src/main/scala/workflow/graph/GatherTransformer.scala

```diff
@@ -0,0 +1,19 @@
+package workflow.graph
+
+import org.apache.spark.rdd.RDD
+
+/**
+ * A [[TransformerOperator]] that gathers multiple datasets of {@tparam T} into a dataset of Seq[T]
+ * (Or individual datums of T into a single Seq[T])
+ */
+private[workflow] case class GatherTransformer[T]() extends TransformerOperator {
```

Comment:
I was confused that this was called `GatherTransformer` mind switching it to `GatherTransformerOperator` - more verbose but its not exactly a regular transformer.